pennylane/transforms/zx/reduce_non_clifford.py

```diff
@@ -0,0 +1,122 @@
+"""
+
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.transforms import transform
+from pennylane.typing import PostprocessingFn
+
+from .converter import from_zx, to_zx
+from .helper import _needs_pyzx
+
+
+@_needs_pyzx
+@transform
+def reduce_non_clifford(tape: QuantumScript) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    """Reduce the number of non-Clifford phase gates applying the full ZX-based pipeline for an arbitrary circuit.
+
+    This transform works performing the following simplification/optimization steps:
+
+        - apply the `full_reduce <https://pyzx.readthedocs.io/en/latest/api.html#pyzx.simplify.full_reduce>`__
+          simplification pipeline to the ``pyzx`` graph representation of the given input circuit;
+
+        - use the `extract_circuit <https://pyzx.readthedocs.io/en/latest/api.html#pyzx.extract.extract_circuit>`__
+          function to extract the equivalent sequence of gates and build a new optimized circuit;
+
+        - apply the `basic_optimization <https://pyzx.readthedocs.io/en/latest/api.html#pyzx.optimize.basic_optimization>`__ pass
+          to further optimize the phase-polynomial blocks circuit.
+
+    This pipeline does not run the Third Order Duplicate and Destroy (TODD) algorithm and thus is not restricted to Clifford + T circuits.
+    The returned circuit is equivalent to the original input up to a global phase.
+
+    Args:
+        tape (QNode or QuantumScript or Callable): the input circuit to be transformed.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumScript], function]:
+        the transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    Raises:
+        ModuleNotFoundError: if the required ``pyzx`` package is not installed.
+
+    **Example:**
+
+    .. code-block:: python3
+
+        import pennylane as qml
+        import pennylane.transforms.zx as zx
+
+        dev = qml.device("default.qubit")
+
+        @zx.reduce_non_clifford
+        @qml.qnode(dev)
+        def circuit(x, y):
+            qml.T(0)
+            qml.Hadamard(0)
+            qml.Hadamard(0)
+            qml.CNOT([0, 1])
+            qml.T(0)
+            qml.RX(x, 1)
+            qml.RX(y, 1)
+            return qml.state()
+
+
+    .. code-block:: pycon
+
+        >>> print(qml.draw(circuit)(3.2, -2.2))
+        0: ──S─╭●─────────────────┤  State
+        1: ────╰X──H──RZ(1.00)──H─┤  State
+
+
+    .. note::
+
+        This transform is designed to minimize non-Clifford phase gates (e.g. T), and it's not as effective
+        at reducing the number of two-qubit gates (e.g. CNOT).
+        However, its performance varies significantly depending on the type of circuit.
+        For example, you might see a substantial increase in CNOT gates when optimizing a circuit composed primarily of Toffoli gates.
+        Conversely, it tends to perform much better on Trotterized chemistry circuits.
```

Comment:
The comparison across different classes of circuits feels a bit strange, but that might just be me. I thought the following might be nicer, isolating the circuit classes a little in the statement.
```suggestion
        Conversely, it tends to perform quite well on Trotterized chemistry circuits.
```